sale_coupon_multi_use/models/sale_coupon.py

```diff
@@ -19,11 +19,14 @@ class SaleCoupon(models.Model):
     # Takes value from related program (coupon_multi_use field), when
     # it is generated.
     multi_use = fields.Boolean(readonly=True)
+    currency_program_id = fields.Many2one(related="program_id.currency_id")
     consumption_line_ids = fields.One2many(
         "sale.coupon.consumption_line", "coupon_id", "Consumption Lines", readonly=True,
     )
-    discount_fixed_amount_delta = fields.Float(
-        "Fixed Amount Delta", compute="_compute_discount_fixed_amount_delta"
+    discount_fixed_amount_delta = fields.Monetary(
+        "Fixed Amount Delta",
+        compute="_compute_discount_fixed_amount_delta",
+        currency_field="currency_program_id",
```

Comment:
To move in your `sale_coupon_multi_currency` module: it's not linked to multi use.

Reply:
This does not make any sense. All `sale_coupon_multi_currency` do is make sale_coupon module convert currencies both ways -> from program to SO and from SO to program.

Actual multi-currency capability is already used on sale_coupon module (standard module. As a matter of fact, all Odoo modules handle currencies where monetary amounts are involved without any special extra modules -> its just hidden if multiple currencies are not used), so it must be on sale_coupon_multi_use. It has nothing to do with `sale_coupon_multi_currency`. If I would move all multi-currency awareness to `sale_coupon_multi_currency` (and glue module), then module would work incorrectly if you you multiple currencies, but not install that module. It would just be broken.

And because `sale_coupon_multi_use` introduces part where it tracks discount from SO, it must convert back to his own currency (if its different).